src/main/java/net/masterthought/cucumber/generators/AbstractPage.java

```diff
@@ -32,30 +12,20 @@
  */
 public abstract class AbstractPage {
 
-    private static final Logger LOG = LogManager.getLogger(AbstractPage.class);
-
-    private final VelocityEngine engine = new VelocityEngine();
-    protected final VelocityContext context = new VelocityContext();
-
     /** Name of the HTML file which will be generated. */
     private final String templateFileName;
-    /** Results of the report. */
-    protected final ReportResult reportResult;
-    /** Configuration used for this report execution. */
-    protected final Configuration configuration;
-
-    protected AbstractPage(ReportResult reportResult, String templateFileName, Configuration configuration) {
+    
+    protected AbstractPage(String templateFileName) {
         this.templateFileName = templateFileName;
-        this.reportResult = reportResult;
-        this.configuration = configuration;
-
-        this.engine.init(buildProperties());
-        buildGeneralParameters();
     }
-
-    public void generatePage() {
-        prepareReport();
-        generateReport();
+    
+    /**
+     * Returns HTML file name (with extension) for this report.
+     *
+     * @return HTML file for the report
+     */
+    public String getTemplateName() {
+        return "templates/generators/" + templateFileName;
```

Comment:
can you leave code better than it is now and move this path into private static field?